### ISSUE TITLE
[bug] git clone for scm attribute fails in pyinstaller package

### ISSUE THREAD

**Issue by ytimenkov** [conan-io/conan#6161]
Conan version 1.20.4

Original problem is that conan wrapped into pyinstaller package (which we use) fails to clone repository for recipe with `scm` attribute:
```
<...>: Configuring sources in /root/.conan/data/.../source
<..>: Getting sources from url: 'ssh://git@bitbucket....'
ERROR: Command 'git clone "ssh://git@bitbucket..." .  ' returned non-zero exit status 128.
```

strace'ing showed that git fails to run because it uses libcrypto from conan (which is ancient) and can't find symbol. Apparently, conan doesn't restore `LD_LIBRARY_PATH` for `git clone` command.

Similar problem was fixed long ago in #3036, but looking at `scm.py` / `run()` it has optional `_runner` which calls just `check_output` without restoring.

If having `_runner` all the time is not an option, at least call to `check_output` should be wrapped into  `pyinstaller_bundle_env_cleaned()` context manager.

It would be nice to go through all calls to `Popen` and check whether they need to be wrapped into this context manager or maybe avoid direct usage of it and have an internal wrapper.

**Comment by memsharded** [comment]
Hi @ytimenkov 

Thanks for the report, yes it makes sense, this would be an issue.
I suggest following your advice and approach this with the following:

1. Do a refactor of the Conan codebase to wrap all subprocess calls. There are few different ways to call it, and it is not very clear how to use it and why. E.g. ``_run_muted``. Pure refactor, no changing in anyway how things execute.
2. Use the ``pyinstaller_bundle_env_cleaned()`` on the abstracted wrapper when necessary.

**Comment by memsharded** [comment]
Did 2 things:

- Refactor of the runners, just to put most of them in a common place
- Added ``pyinstaller_bundle_env_cleaned()`` to the SCM one, which is the one with the bug. The others are most likely not affected by this.

Merged, will be released in Conan 1.22.
